js/main.js

```diff
@@ -0,0 +1,146 @@
+import { getNextSlide, getPrevSlide, getSlideActiveDot } from "../js/script.js";
+
+function Slider(container) {
+  this.container = document.querySelector(container);
+  this.slides = Array.from(this.container.querySelectorAll(".slide"));
+  this.buttons = this.container.querySelectorAll(".buttons div");
+  this.dotNode = this.container.querySelector(".dots");
+  let timeOutId;
+  let that = this;
```

Comment:
tại sao lại gần gán như này anh nhỉ?

Reply:
Do this ở trong một số hàm nó không trỏ đến function constructor như mình mong muốn, nên muốn trỏ đến đó mình cần khai báo một biến that để dùng thay thế mà vẫn trỏ được đến function constructor.